app/assets/javascripts/components/common/enroll_button.jsx

```diff
@@ -1,228 +1,225 @@
+    }
+    if (!Features.wikiEd) {
+      const requestedAccountsUrl = `/requested_accounts/${course.slug}`;
+      requestedAccountsLink = <p key="requested_accounts"><a href={requestedAccountsUrl}>{I18n.t('courses.requested_accounts')}</a></p>;
     }
 
-    let buttonClass = 'button';
-    buttonClass += this.props.inline ? ' border plus' : ' dark';
-    const buttonText = this.props.inline ? '+' : CourseUtils.i18n('enrollment', this.props.course.string_prefix);
-
-    // Remove this check when we re-enable adding users by username
-    const button = (
-      <button
-        className={buttonClass}
-        onClick={() => {
-          this.props.open();
-          setTimeout(() => {
-            this.refs.username.focus();
-          }, 125);
-        }}
-      >
-        {buttonText}
-      </button>
+    editRows.push(
+      <tr className="edit" key="enroll_students">
+        <td>
+          <p>{I18n.t('users.course_passcode')}&nbsp;<b>{course.passcode}</b></p>
+          <p>{I18n.t('users.enroll_url')}</p>
+          <input type="text" readOnly={true} value={enrollUrl} style={{ width: '100%' }} />
+          {massEnrollmentLink}
+          {requestedAccountsLink}
+        </td>
+      </tr>
     );
+  }
 
-    return (
-      <div className="pop__container" onClick={this.stop}>
-        {button}
-        <Popover
-          is_open={this.props.is_open}
-          edit_row={editRows}
-          rows={users}
-        />
-      </div>
+  // This row allows permitted users to add usrs to the course by username
+  // @role controls its presence in the Enrollment popup on /students
+  // @allowed controls its presence in Edit Details mode on Overview
+  if (role === 0 || allowed) {
+    // Instructor-specific extra fields
+    let realNameInput;
+    let roleDescriptionInput;
+    if (role === 1) {
+      realNameInput = <input type="text" ref={realNameRef} placeholder={I18n.t('users.name')} />;
+      roleDescriptionInput = <input type="text" ref={roleDescriptionRef} placeholder={I18n.t('users.role.description')} />;
+    }
+
+    editRows.push(
+      <tr className="edit" key="add_students">
+        <td>
+          <form onSubmit={enroll}>
+            <input type="text" ref={usernameRef} placeholder={I18n.t('users.username_placeholder')} />
+            {realNameInput}
+            {roleDescriptionInput}
+            <button className="button border" type="submit">{CourseUtils.i18n('enroll', course.string_prefix)}</button>
+          </form>
+        </td>
+      </tr>
     );
   }
-}
-);
-
-const mapDispatchToProps = { initiateConfirm, addNotification, addUser, removeUser };
 
-export default withRouter(connect(null, mapDispatchToProps)(
-  Conditional(PopoverExpandable(EnrollButton))
-));
+  let buttonClass = 'button';
+  buttonClass += inline ? ' border plus' : ' dark';
+  const buttonText = inline ? '+' : CourseUtils.i18n('enrollment', course.string_prefix);
+
+  // Remove this check when we re-enable adding users by username
```

Comment:
This looks like an outdated comment that can be removed.

Reply:
Ok, done.